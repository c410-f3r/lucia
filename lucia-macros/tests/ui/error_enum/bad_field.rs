#[lucia_macros::pkg(api(Foo), data_format(json))]
mod pkg {
  #[derive(Debug)]
  #[pkg::req_data]
  pub struct Req(#[pkg::bar] i32);

  #[pkg::res_data]
  struct Res;
}

#[lucia_macros::pkg(api(Foo), data_format(json))]
mod pkg {
  #[derive(Debug)]
  #[pkg::req_data]
  pub struct Req(#[pkg::field(foo = "1111")] i32);

  #[pkg::res_data]
  struct Res;
}

fn main() {
}
