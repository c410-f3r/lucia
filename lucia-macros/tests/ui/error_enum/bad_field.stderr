error: Field attributes must be annotated as `#[pkg::field(name = "SomeName")]`
 --> tests/ui/error_enum/bad_field.rs:5:20
  |
5 |   pub struct Req(#[pkg::bar] i32);
  |                    ^^^

error: expected `name`
  --> tests/ui/error_enum/bad_field.rs:15:31
   |
15 |   pub struct Req(#[pkg::field(foo = "1111")] i32);
   |                               ^^^
